pkg/skaffold/deploy/deploy_mux.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deploy
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build"
+)
+
+// DeployerMux forwards all method calls to the deployers it contains.
+// When encountering an error, it aborts and returns the error. Otherwise,
+// it collects the results and returns it in bulk.
+type DeployerMux []Deployer
+
+var _ Deployer = DeployerMux{}
+
+func (m DeployerMux) Labels() map[string]string {
+	labels := make(map[string]string)
+	for _, deployer := range m {
+		copyMap(labels, deployer.Labels())
+	}
+	return labels
+}
+
+func (m DeployerMux) Deploy(ctx context.Context, w io.Writer, as []build.Artifact, ls []Labeller) *Result {
+	seenNamespaces := sets.String{}
```

Comment:
this is cool, but kinda scares me that it's in kubernetes. Haven't made up my mind yet about it. 
The risk of it disappearing / changing is minimal as skaffold is for kubernetes. But in general we tend to avoid depending on utilities from dependencies - unless the dependency is specifically for that utility. E.g. docker has some great stuff in their ioutils package but we decided not to use them.

Reply:
Yeah that totally makes sense. The de-duplication is easily implemented so there's no need to add a dependency here.
I think there are a few places where Skaffold code could benefit from a StringSet util, but this is not the right PR to make that change :). So I've inlined a small helper in this file.